src/backend/distributed/executor/insert_select_executor.c

```diff
@@ -210,8 +210,31 @@ CoordinatorInsertSelectExecScanInternal(CustomScanState *node)
 							 distSelectJob->jobId);
 			char *distResultPrefix = distResultPrefixString->data;
 
-			CitusTableCacheEntry *targetRelation =
+			CitusTableCacheEntry *cachedTargetRelation =
 				GetCitusTableCacheEntry(targetRelationId);
+			CitusTableCacheEntry *targetRelation = palloc(sizeof(CitusTableCacheEntry));
+			*targetRelation = *cachedTargetRelation;
+
+#if PG_USE_ASSERT_CHECKING
+			targetRelation->partitionKeySTring = NULL;
+			targetRelation->shardIntervalCompareFunction = NULL;
+			targetRelation->hashFunction = NULL;
+			targetRelation->arrayOfPlacementArrayLengths = NULL;
+			targetRelation->arrayOfPlacementArrays = NULL;
+			targetRelation->referencedRelationViaForeignKey = NULL;
+			targetRelation->referencingRelationsViaForeignKey = NULL;
+#endif
+			targetRelation->partitionColumn = copyObject(
+				cachedTargetRelation->partitionColumn);
+			targetRelation->sortedShardIntervalArray =
+				palloc(targetRelation->shardIntervalArrayLength * sizeof(ShardInterval));
+			for (int shardIndex = 0; shardIndex <
+				 targetRelation->shardIntervalArrayLength; shardIndex++)
+			{
+				targetRelation->sortedShardIntervalArray[shardIndex] =
+					CopyShardInterval(
+						cachedTargetRelation->sortedShardIntervalArray[shardIndex]);
+			}
 
```

Comment:
Why are we not copying `partitionKeyString`, `shardIntervalCompareFunction`, `hashFunction`,  and other members that are freed at `ResetCitusTableCacheEntry`? Is that because we are sure we don't use them?